### ISSUE TITLE
 'MutableRefObject<HTMLDivElement | undefined>' is not assignable to type 'RefObject<FocusableElement>

### ISSUE THREAD

**Issue by hhimanshu** [chakra-ui/chakra-ui#4953]
### Description

When I use `initialFocusRef` using Typescript code, it fails the compilation

### Link to Reproduction

https://codesandbox.io/s/sweet-cloud-uvr0u?file=/src/components/FilterModal/index.tsx

### Steps to reproduce

1. Click on the code sandbox link and you will see the compilation failure

### Chakra UI Version

1.6.10

### Browser

Firefox Developer Edition 94.0b9 (64-bit)

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

What I am looking for is
1. When I click on "All Nutrients" button, the modal opens and focuses on the section which says "Nutrients"

Thank you

**Comment by segunadebayo** [comment]
Hi @hhimanshu,

Here are some issues I found with the components:

- You can't pass `ref` as a prop. You'll need to use `forwardRef` if you want that.
- To make a  `div` element focusable, you'll need to add `tabIndex=-1` to it.

Here's the working solution: https://codesandbox.io/s/musing-colden-89ose?file=/src/components/FilterModal/NutrientFilterOptions.tsx

Hope you find this useful.

Cheers mate.

**Comment by hhimanshu** [comment]
Hey @segunadebayo,
Thank you so much for your response. Regarding your comments

> You can't pass ref as a prop. You'll need to use forwardRef if you want that.

I see that in the link you provided, you have used `useRef` and passed it as a prop, and not `forwardRef`. And if I understand correctly, [useRef](https://reactjs.org/docs/hooks-reference.html#useref) returns a `MutableRefObject` and not `forwardRef`. Am I missing something?

> To make a div element focusable, you'll need to add tabIndex=-1 to it.

Is this documented anywhere? 

Please let me know, and THANK YOU again for your help

**Comment by hhimanshu** [comment]
Hello @segunadebayo, I made another attempt by learning about `ForwardRef`, however that does not compile at all. The reproducible link is available at https://codesandbox.io/s/busy-fog-n3vcb?file=/src/components/FilterModal/index.tsx:1128-1162

I also had to make sue of `FocusableElement` from `@chakra-ui/utils` as the code was complaining about it.

**Comment by hhimanshu** [comment]
Hello @segunadebayo , I made another attempt that passes `ref` as a prop instead of using `ForwardRef`. Did you mean this in your first comment?

To make it compile with TypeScript I had to import `FocusableElement` from `@chakra-ui/utils`. My current attempt seems to work and the codesandbox link is available at https://codesandbox.io/s/nifty-cannon-1u6r0?file=/src/App.tsx

Let me know if this is the approach you were talking about.
Also, I am still interested in learning more about `tablndex={-1}` trick
